openassessment/assessment/models/staff.py

```diff
@@ -0,0 +1,117 @@
+"""
+Models for managing staff assessments.
+"""
+from datetime import timedelta
+
+from django.db import models, DatabaseError
+from django.utils.timezone import now
+
+from openassessment.assessment.models.base import Assessment
+from openassessment.assessment.errors import StaffAssessmentInternalError
+
+
+class StaffWorkflow(models.Model):
+    """Internal Model for tracking Staff Assessment Workflow
+
+    This model can be used to determine the following information required
+    throughout the Staff Assessment Workflow:
+
+    1) Get next submission that requires assessment.
+    2) Does a submission have a staff assessments?
+    4) Does this staff member already have a submission open for assessment?
+    5) Close open assessments when completed.
+
+    """
+    # Amount of time before a lease on a submission expires
+    TIME_LIMIT = timedelta(hours=8)
+
+    scorer_id = models.CharField(max_length=40, db_index=True)
+    course_id = models.CharField(max_length=40, db_index=True)
+    item_id = models.CharField(max_length=128, db_index=True)
+    submission_uuid = models.CharField(max_length=128, db_index=True, unique=True)
+    created_at = models.DateTimeField(default=now, db_index=True)
+    grading_completed_at = models.DateTimeField(null=True, db_index=True)
+    grading_started_at = models.DateTimeField(null=True, db_index=True)
+    cancelled_at = models.DateTimeField(null=True, db_index=True)
+    assessment = models.CharField(max_length=128, db_index=True, null=True)
+
+    class Meta:
+        ordering = ["created_at", "id"]
+        app_label = "assessment"
+
+    @property
+    def is_cancelled(self):
+        """
+        Check if the workflow is cancelled.
+
+        Returns:
+            True/False
+        """
+        return bool(self.cancelled_at)
+
+    @classmethod
+    def get_submission_for_review(cls, course_id, item_id, scorer_id):
+        """
+        Find a submission for peer assessment. This function will find the next
+        submission that requires assessment, excluding any submission that has been
+        completely graded, or is actively being reviewed by other students.
```

Comment:
Nit: a comment above says that it won't return the staff member's own submissions either. Is that an issue? It might be good to disallow staff from reviewing their own submissions, but I think Leslie said she wasn't concerned about that.